src/Mono.Android/Java.Interop/JavaObjectExtensions.cs

```diff
@@ -122,10 +122,12 @@ static IJavaObject CastClass (IJavaObject instance, Type resultType)
 							instance.GetType ().FullName, resultType.FullName));
 		}
 
-		// typeof(Foo) -> FooSuffix
-		// typeof(Foo<>) -> FooSuffix`1
-		internal static Type? GetHelperType (Type type, string suffix)
+		// typeof(Foo) -> FooInvoker
+		// typeof(Foo<>) -> FooInvoker`1
+		[UnconditionalSuppressMessage ("Trimming", "IL2026", Justification = "*Invoker types are preserved by the MarkJavaObjects linker step.")]
+		internal static Type? GetInvokerType (Type type)
```

Comment:
@jonpryor this was the part, I wanted to check it's OK to rename this method.

I thought allowing any `string suffix` through here could introduce a future linker issue, because `MarkJavaObjects` only preserves `*Invoker` and `*Adapter` right now.

Reply:
This just raises a related question: what's a `*Adapter` type, and why is it being preserved?  I forget. :-(

I think for "consistency" we should consider updating `src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs` to likewise rename  `GetHelperType()` to, or otherwise introduce, `GetInvokerType()`.  At least then our internal terminology would be consistent.

Reply:
This appears to be the commit which introduce the `Adapter` suffix; it was apparently for interfaces: https://github.com/xamarin/monodroid/commit/ff043c0281fbe1fefb3e83b7ccda1eca791c3fb0

Reply:
I suspect what happened is we "unified" the `Adapter` and `Invoker` suffixes into `Invoker`, but I can't easily find such a commit.

Reply:
Looks like `Adapter` was removed in `generator` output, with `Invoker` as the replacement, as part of https://github.com/xamarin/monodroid/commit/efeff037dfc4d4c60ad87fd6d6b22d9af45d7583

So it looks like there's no need to preserve `Adapter`-suffixed types, not since 2011.

Reply:
Updated the `MarkJavaObjects` step. I wonder if we'll see any `.apk` size differences?